### PR TITLE
Allow query params on all custom routes

### PR DESCRIPTION
**Pull request by BrennerSpear**

### CODE REVIEW

**Comment by BrennerSpear** [comment]
Addresses issues #503, half of #544, and fixes what PR #531 was trying to fix

File: test/server/plural.js [hunk 1]
```diff
@@ -710,27 +712,38 @@ describe('Server', () => {
...
       request(server)
         .get('/comments/special/1-quux')
         .expect([db.comments[4]])
     ))
 
+    it('should rewrite using params and client-defined query', () => (
+      request(server)
+        .get('/blog/comments?userId=1')
+        .expect([db.comments[0],db.comments[2],db.comments[4]])
+    ))
```
Comment (BrennerSpear) [inline test/server/plural.js]: This test would fail without the change in ```rewriter.js```

**Comment by BrennerSpear** [comment]
Previously a custom route like ```'/blog/:resource' : '/:resource'``` that had a request come in that looked like ```/blog/comments?userId=1``` Would ignore the query parameters of the request and return all comments. This fix will turn the correctly filtered comments.

**Comment by typicode** [comment]
@BrennerSpear thank you, the PR looks great but I've preferred to use a third-party package to handle URL rewriting instead.

Mainly to reduce maintainance on this part of JSON Server.

Query params in custom routes are handled also by this package.

**Pull request closed without merging**